src/main/scala/com/microsoft/partnercatalyst/fortis/spark/transforms/locations/client/FeatureServiceClient.scala

```diff
@@ -23,21 +26,37 @@ class FeatureServiceClient(host: String) extends Serializable {
   private def parseResponse(response: String): Iterable[FeatureServiceFeature] = {
     implicit val formats = json.DefaultFormats
 
-    json.parse(response).extract[FeatureServiceResponse].features
+    try {
+      json.parse(response).extract[FeatureServiceResponse].features
+    } catch {
+      case ex: ParseException =>
+        logError(s"Unable to parse feature service response: $response", ex)
+        List()
```

Comment:
As a client of this code, I'd like to distinguish between the error case and having no features found. Perhaps this interface would be a good candidate for a return type of Try[T]?

[Here's a pretty good article](http://danielwestheide.com/blog/2012/12/26/the-neophytes-guide-to-scala-part-6-error-handling-with-try.html) describing functional error handling with Try in Scala.